velox/exec/Window.cpp

```diff
@@ -573,9 +615,21 @@ vector_size_t Window::callApplyLoop(
 
   // This function requires that the currentPartition_ is available for output.
   VELOX_DCHECK_NOT_NULL(currentPartition_);
+
+  // Used to record the processed row count in current partition. And then
+  // passed this value to clearOutputRows to clear the real row count.
+  auto processedRowCount = -1;
   while (numOutputRowsLeft > 0) {
-    auto rowsForCurrentPartition =
-        currentPartition_->numRows() - partitionOffset_;
+    auto totalRows = currentPartition_->numRowsForProcessing();
```

Comment:
Nit : This can be written in a more compact way as 
auto rowsForCurrentPartition = currentPartition_->isPartial() ? currentPartition_->numRowsForProcessing() : currentPartition_->numRowsForProcessing() - partitionOffset;